### ISSUE TITLE
Zoom to feature showing incorrect feature 

### ISSUE THREAD

**Issue by jjoseph174** [qgis/QGIS#38226]
**Describe the bug**
After opening attribute table and when clicking "zoom to feature" on a particular row, the feature that shows up is no longer accurate and not associated with the address and owner that is displayed in the attribute table. This issue recently started happening and it is now showing an unrelated feature. My initial thoughts were after sorting the columns by order the data got mixed up but this shouldn't happen. For example when using "Identify Features" on the point the owner should be "Rosenfeld concrete corp" but the attribute table incorrectly associates the owner "Taylor Lisa.." with this feature. 
<img width="817" alt="ex" src="https://user-images.githubusercontent.com/69473692/89803615-10b57880-db01-11ea-8cb1-065627f919b1.PNG">
  -->

**How to Reproduce**

Steps, sample datasets and qgis project file to reproduce the behavior. Screencasts or screenshots welcome

1. Go to attribute table
2. Click on zoom to feature
3. Feature shown does not correspond with what the attribute table says (ex. address in attribute table is not the address shown in the map)

**QGIS and OS versions**

QGIS version
3.10.5-A Coruña
QGIS code revision
984615fe1e
Compiled against Qt
5.11.2
Running against Qt
5.11.2
Compiled against GDAL/OGR
3.0.4
Running against GDAL/OGR
3.0.4
Compiled against GEOS
3.8.1-CAPI-1.13.3
Running against GEOS
3.8.1-CAPI-1.13.3
Compiled against SQLite
3.29.0
Running against SQLite
3.29.0
PostgreSQL Client Version
11.5
SpatiaLite Version
4.3.0
QWT Version
6.1.3
QScintilla2 Version
2.10.8
Compiled against PROJ
6.3.1
Running against PROJ
Rel. 6.3.1, February 10th, 2020
OS Version
Windows 10 (10.0)
Active python plugins
quick_map_services; 
valuetool; 
db_manager; 
MetaSearch; 
processing

**Comment by gioman** [comment]
@jjoseph174 please add a sample project+data.

**Comment by jjoseph174** [comment]
[MA_QGIS_Storage_Milford_Hopedale_Bellingham_Mendon.zip](https://github.com/qgis/QGIS/files/5052293/MA_QGIS_Storage_Milford_Hopedale_Bellingham_Mendon.zip)

Here is a a sample project. The attribute table data is mixed up for the following layers (BHMM_BA_with_Coordinates
& BHMM_All_BA_Grtrthn2Acres_With_Slope_2miSub_Buffer). However the parcel layer attribute table shows the correct owner info and "zoom to feature" zooms to correct data.

**Comment by gioman** [comment]
> [MA_QGIS_Storage_Milford_Hopedale_Bellingham_Mendon.zip](https://github.com/qgis/QGIS/files/5052293/MA_QGIS_Storage_Milford_Hopedale_Bellingham_Mendon.zip)

@jjoseph174 there is no data in zip file, only the project file.

**Comment by uclaros** [comment]
When was 3.10.5? I think this has been reported and fixed already.

**Comment by jjoseph174** [comment]
How do I add data? Do you know how to fix it?

**Comment by uclaros** [comment]
Can you also try with the latest 3.10 LTR version?

**Comment by gioman** [comment]
> How do I add data? Do you know how to fix it?

@jjoseph174 zip it together with your project. Try 3.10.8 before.

**Comment by jjoseph174** [comment]
Just downloaded latest 3.10.8 version. Zoom to feature still zooms to incorrect data. How do I export the data and zip it in the project?

**Comment by gioman** [comment]
> How do I export the data and zip it in the project?

@jjoseph174 you don't need to send over all the layers that are in your project, just send a small sample that demonstrates the issue you described.

Create a new folder, copy in there the datasource/s you need (I see you are using shapefiles). Then create a new project add data from the folder, and save the project in the same folder. Zip it and attach it here.

**Comment by jjoseph174** [comment]
@gioman Okay thanks for explaining I believe I have everything here. When zooming to the feature listed in the attribute table it still doesn't correspond to the site address in the attribute table. 
[MA_QGIS.zip](https://github.com/qgis/QGIS/files/5052890/MA_QGIS.zip)

**Comment by roya0045** [comment]
Wait, so the data returned by the identify feature tool is wrong also or am I misreading something?

**Comment by uclaros** [comment]
> Wait, so the data returned by the identify feature tool is wrong also or am I misreading something?


Apparently yes. QGIS works fine, the issue is on the data. Judging from the field names I'd say that the two `BHMM_BA` layers had their attributes joined from some table on the wrong field!

**Comment by jjoseph174** [comment]
Yes, when using the identifying feature and zoom to feature function, the data shown is only correct for the BHMM_Parcels layer. For the other 2 layers when using the identifying feature and zoom to feature function, it is zooming into the wrong parcel data. It worked last week and would zoom into the correct & corresponding data/address/owner info in the attribute table and map. I'm not sure how the data got mixed up (it's now showing the wrong parcel for the data in the attribute table and vice versa) and would like to know if it is a bug and can be fixed somehow.

**Comment by gioman** [comment]
> For example when using "Identify Features" on the point the owner should be "Rosenfeld concrete corp" but the attribute table incorrectly associates the owner "Taylor Lisa.." with this feature.



> QGIS works fine, the issue is on the data.

@jjoseph174 
I agree. I just tested the data, exactly in the same spot as the screenshot in the description. The problem is in the data, not QGIS.

**Comment by gioman** [comment]
> and would like to know if it is a bug and can be fixed somehow.

@jjoseph174 is not a bug in QGIS, the problem is in your data.

**Comment by jjoseph174** [comment]
@gioman I see. I'm unsure why for example if the attribute table lists the address as 100 Deer St and if I zoom to feature on that row, it zooms to a parcel which is not near that address. It used to zoom to the correct address/parcel. It's because of the data?

**Comment by gioman** [comment]
> It's because of the data?

@jjoseph174 yes, your point data is not matching your other layers, but this is not QGIS fault, the data is just wrong.

**Comment by jjoseph174** [comment]
@gioman Okay thanks everyone for helping I will look into this further!

**Comment by DelazJ** [comment]
@jjoseph174 Maybe something to do with
>  My initial thoughts were after **sorting the columns by order** the data got mixed up but this shouldn't happen.

How did the data get sorted?

**Comment by jjoseph174** [comment]
@DelazJ The data in the attribute table was sorted just by clicking the field name in the top of the columns which sorted it by alphabetical order/ascending order. I thought maybe since I was clicking the different column names to sort alphabetically by address or owner name the data got mixed up and now with zoom to feature it is showing the wrong parcel/address shown in the attribute table

**Comment by DelazJ** [comment]
> by clicking the field name in the top of the columns

Ah OK. I was afraid of some .dbf file edit in Calc/Excel which would certainly have corrupted the data.
But in this case, it's unlikely that sorting the way you did would corrupt the data. This is just for visualizing and does not edit your data.

**Comment by jjoseph174** [comment]
@DelazJ I see that makes sense. It definitely is strange that it did work a week ago but after revisiting the project I noticed after clicking "zoom to feature" for a particular row in the attribute table it would zoom to a different address (which is also listed in the attribute table). Like in the screenshot I provided, for the data on Main St shown in the identify results/attribute table it zooms to an address on Plain St which is incorrect/wrongly associated. And was curious if anyone knows how to fix it
